### DISCUSSION TITLE
Basic templating via yaml

### DISCUSSION THREAD

**Discussion by mhf-ir** [hairyhenderson/gomplate#1019] [Q&A] [answered]
have config file.

‍‍
```yml
hosts:
  - ali
  - mamad
```
‍
And template file:

```
{{ range $sn := $hosts }}
server {
  server_name {{ $sn }};
}
{{ end }}
```
‍

How can i run gomplate for generate:

```
server {
  server_name ali;
}
server {
  server_name mamad;
}
```
‍

BTW it's awesome project but need more examples for simples use cases.

**Comment by hairyhenderson** [comment, marked as answer]
Hi @mhf-ir,

First off, thanks for your kind words - and I agree more simple examples would help - perhaps something you could be willing to contribute?

To use a YAML file as a data source you'll need to use either the [`--datasource`](https://docs.gomplate.ca/usage/#datasource-d) or [`--context`](https://docs.gomplate.ca/usage/#context-c) flags, or `-d`/`-c` for short. Personally I prefer the `--context`/`c` approach when possible (though you can't use it if the data source needs to be read dynamically).

So, you could run a command like this: `gomplate -c config=config.yaml -f input.tmpl`

In your template file instead of using `$hosts` (which hadn't been defined yet), you'll need to reference [the context](https://docs.gomplate.ca/syntax/#the-context):

```go
{{ range $sn := .config.hosts }}
server {
  server_name {{ $sn }};
}
{{ end }}
```

And putting that all together:

```console
$ gomplate -c config=config.yaml -f input.tmpl

server {
  server_name ali;
}

server {
  server_name mamad;
}
```

Let me know if that helps!
